src/execution/index/art/art.cpp

```diff
@@ -272,55 +274,56 @@ static void ConcatenateKeys(ArenaAllocator &allocator, Vector &input, idx_t coun
 		}
 
 		// Concatenate the keys.
-		auto other_key = ARTKey::CreateARTKey<T>(allocator, input_data[idx]);
-		keys[i].Concat(allocator, other_key);
+		auto other_key = ARTKey::CreateARTKey<T>(allocator, input_data[idx], max_len);
+		keys[i].Concat(allocator, other_key, max_len);
 	}
 }
 
 template <bool IS_NOT_NULL>
-void GenerateKeysInternal(ArenaAllocator &allocator, DataChunk &input, unsafe_vector<ARTKey> &keys) {
+void GenerateKeysInternal(ArenaAllocator &allocator, DataChunk &input, unsafe_vector<ARTKey> &keys,
+                          const idx_t max_len) {
 	switch (input.data[0].GetType().InternalType()) {
 	case PhysicalType::BOOL:
-		TemplatedGenerateKeys<bool, IS_NOT_NULL>(allocator, input.data[0], input.size(), keys);
+		TemplatedGenerateKeys<bool, IS_NOT_NULL>(allocator, input.data[0], input.size(), keys, max_len);
 		break;
 	case PhysicalType::INT8:
-		TemplatedGenerateKeys<int8_t, IS_NOT_NULL>(allocator, input.data[0], input.size(), keys);
+		TemplatedGenerateKeys<int8_t, IS_NOT_NULL>(allocator, input.data[0], input.size(), keys, max_len);
```

Comment:
Do we need to pass this as a parameter into all of these functions? Can we not create the art keys first and then later check if they have exceeded the maximum length?

Reply:
Sounds good!